modules/apm/src/main/java/METERING.md

```diff
@@ -0,0 +1,171 @@
+
+## How to use an instrument
+There are 2 types of usages of an instrument depending on a type.
+- For synchronous instrument (counter/UpDownCounter) we need to register an instrument with
+  `MeterRegistry` and use the returned value to increment a value of that instrument
+```java
+  MeterRegistry registry;
+  LongCounter longCounter = registry.registerLongCounter("es.test.requests.count", "a test counter", "count");
+  longCounter.increment();
+  longCounter.incrementBy(1, Map.of("name", "Alice"));
+  longCounter.incrementBy(1, Map.of("name", "Bob"));
+```
+
+- For asynchronous instrument (gauge/AsynchronousCounter) we register an instrument
+  and have to provide a callback that will report the absolute measured value.
+  This callback has to be provided upon registration and cannot be changed.
+```java
+MeterRegistry registry;
+long someValue = 1;
+registry.registerLongGauge("es.test.cpu.temperature", "a test gauge", "celcius",
+() -> new LongWithAttributes(someValue, Map.of("cpuNumber", 1)));
+```
+
+If we don’t have access to ‘state’ that will be fetched on metric event (when callback is executed)
+we can use a utility LongGaugeMetric or LongGaugeMetric
+```java
+MeterRegistry meterRegistry ;
+LongGaugeMetric longGaugeMetric = LongGaugeMetric.create(meterRegistry, "es.test.gauge", "a test gauge", "total value");
+longGaugeMetric.set(123L);
+```
+### The use of attributes aka dimensions
+Each instrument can attach attributes to a reported value. This helps drilling down into the details
+of value that was reported during the metric event
+
+
+## Development
+
+### Fake http server
+
+The quickest way to verify that your metrics are working is to run `./gradlew run --with-apm-server`.
+This will run ES node (or nodes in serverless) and also start a fake http server that will act
+as an apm server. This fake http server will log all the http messages it receives from apm-agent
+
+### With APM server in cloud
+You can also run local ES node with an apm server in cloud.
+Create a new deployment in cloud, then click the 'hamburger :)' on the left, scroll to Observability and click APM under it.
+At the upper right corner there is `Add data` link, then scroll down to `ApmAgents` section and pick Java
+There you should be able to see `elastic.apm.secret_token` and `elastic.apm.server_url. You will use them in the next step.
+
+Next you should create a file `apm_server_ess.gradle`
+in a different directory than your elasticsearch checkout (so that branch changes don't remove it)
+The content of the file:
+```
+rootProject {
+  if (project.name == 'elasticsearch') {
+    afterEvaluate {
+      testClusters.matching { it.name == "runTask" }.configureEach {
+        setting 'xpack.security.audit.enabled', 'true'
+        keystore 'tracing.apm.secret_token', 'REDACTED'
+        setting 'telemetry.metrics.enabled', 'true'
+
+      setting 'tracing.apm.agent.server_url', 'https://REDACTED:443'
+      }
+    }
+  }
+}
+```
```

Comment:
Let's just removed this and suggest the init.d/apm.gradle to reduce confusion.

Reply:
done